packages/api/src/modules/transaction/services.ts

```diff
@@ -369,10 +370,17 @@ export class TransactionService implements ITransactionService {
     const api_transaction = await this.findById(bsafe_txid);
     const { predicate, txData, witnesses } = api_transaction;
     const provider = await Provider.create(predicate.provider);
-    const _witnesses = witnesses
+    let _witnesses = witnesses
       .filter(w => !!w.signature)
       .map(witness => witness.signature);
 
+    if (txData.type === TransactionType.Create) {
+      _witnesses = [
+        hexlify(txData.witnesses[txData.bytecodeWitnessIndex]),
+        ..._witnesses,
+      ];
+    }
+
```

Comment:
Change to get the type of transaction by `Transaction.type`
https://github.com/infinitybase/bako-safe-api/blob/5464e3e84448272c7bc156d7cc5b6b91c6be7612/packages/api/src/models/Transaction.ts#L18